### PR TITLE
Simplify filters

### PR DESCRIPTION
**Pull request by afogel**

### CODE REVIEW

File: app/assets/javascript/ahoy_captain/controllers/search_select_controller.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    query: String,
+    url: String,
+    selected: Array
+  }
+
+  connect() {
+    this.search = this.search.bind(this)
+    this.select = new SlimSelect({
+      select: this.element,
+      data: [],
```
Comment (afogel) [inline app/assets/javascript/ahoy_captain/controllers/search_select_controller.js:15]: do we want to allow preloading data, or should we always start from scratch?

File: app/assets/javascript/ahoy_captain/controllers/filter_controller.js [hunk 2]
```diff
@@ -1,145 +0,0 @@
...
-    const mergedData = predicates.map(predicate => {
-      const matchingFilter = filterValues.find(filter => filter.name === predicate.name);
-      if (matchingFilter.selections.length > 0) {
-        return {...predicate, ...matchingFilter}
-      }
-    }).filter(el => el !== undefined);
-    return mergedData;
-  }
-
-  resetFilters(event) {
```
Comment (afogel) [inline app/assets/javascript/ahoy_captain/controllers/filter_controller.js:112]: this refactor breaks resetFilters

File: app/assets/javascript/ahoy_captain/controllers/filter_controller.js [hunk 3]
```diff
@@ -1,145 +0,0 @@
...
-    this.applyFilters(event)
-  }
-
-  applyFilters(e) {
-    e.preventDefault();
-    const searchParams = new URLSearchParams(window.location.search);
-    const filters = this.#filtersForQuery();
-    filters.forEach(filter => {
-      const name = `q[${filter.column_predicate}][]`
-      const selectedValues = searchParams.getAll(name)
```
Comment (afogel) [inline app/assets/javascript/ahoy_captain/controllers/filter_controller.js:125]: this change also breaks conditional filtering, e.g., if I selected CA as a country, I should only be shown regions in Canada

Comment (afogel) [reply app/assets/javascript/ahoy_captain/controllers/filter_controller.js:125]: nvm, it's already broken.

Comment (joshmn) [reply app/assets/javascript/ahoy_captain/controllers/filter_controller.js:125]: resolved

**Review by joshmn — APPROVED** [review]

**Pull request merged**